### ISSUE TITLE
Learning Rate finder for optimizers

### ISSUE THREAD

**Issue by GereonFranken** [tensorflow/tensorflow#43183]
**System information**
- TensorFlow version: 2.3
- Are you willing to contribute it: Yes

**Describe the feature and the current behavior/state.**
--
I am currently reading the book "Deep Learning" by Ian Goodfellow, Yoshua Bengio and Aaron Courville. In this book they propose a formula for finding the best learning rate  under certain constraints. The formula is: 
 `e* = g.T*g / g.T*H*g`
where g is the gradient and H is the Hessian matrix at X(0).
Out of interest i implemented a short script for a random loss function with two variables x,y and a random data point X(0) = {x:1,y:1}. This gives me in this arbitrary example an optimal learning rate of `0.171232876712329`.
[OptimalLR.pdf](https://github.com/tensorflow/tensorflow/files/5213736/OptimalLR.pdf)

Unfortunately i do not have the in depth tensorflow knowledge to play with lets say the Adam Optimizer to try it out there. One would need to get the loss function, take a data point (i assume mini batches work as well) and apply the formula to it and see if it evaluates a valid learning rate. Then run test with the default 0.01 learning rate versus the calculated learning rate and see if the calculated one performs significantly better which would justify the extra calculation. 

I would apprechiate feedback on whether that is possible or whether i am missing something obvious as a reason why that is not possible. For example is this calculation even feasible for deep neural networks?

**Who will benefit with this feature?**
--

If this algorithm is possible one could replace the fixed default learning rate of 0.01 to a calculated one which potentially improves the performance of neural networks regarding the loss function.

**Comment by bhack** [comment]
Please open this support questions to stackoverflow.
In the meantime you can quickly control the learning rate at https://www.tensorflow.org/guide/keras/custom_callback?hl=en#learning_rate_scheduling

**Comment by Saduf2019** [comment]
@GereonFranken 
Please update as per above comment.

**Comment by GereonFranken** [comment]
Will do thank you. Should i leave this issue open or close it then?

**Comment by Saduf2019** [comment]
@GereonFranken 
Thank you for your response, please move the issue to closed status as it will be tracked at stackoverflow.
